src/components/Map/index.tsx

```diff
@@ -26,16 +32,44 @@ export const Map: FC<MapProps> = ({
   children,
   ...otherMapProps
 }) => {
+  const { pathname, query, replace } = useRouter()
+  const mappedQuery = mapRawQueryToState(query)
+  const transitionProps = {
+    transitionDuration: 2000,
+    transitionEasing: (t: number) =>
+      t < 0.5 ? 4 * t * t * t : (t - 1) * (2 * t - 2) * (2 * t - 2) + 1,
```

Comment:
Wow, what is this magic? 😄

Reply:
Hahaha it is a copy/pasted easeInOutQuad function. I will extract it and give it a name. :D